_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -70,12 +87,52 @@ proc _cmake_get_build_dir {} {
     return [option worksrcpath]
 }
 
-default configure.dir {[_cmake_get_build_dir]}
-
-pre-configure {
-    file mkdir ${configure.dir}
+option_proc cmake.generator handle_generator
+proc handle_generator {option action args} {
+    global cmake.generator destroot destroot.target build.cmd build.post_args depends_build destroot.post_args
+    if {${action} eq "set"} {
+        switch -nocase ${args} {
+            "{Unix Makefiles}" {
+                ui_debug "Selecting the 'Unix Makefiles' generator"
+                cmake.generator     "Unix Makefiles"
+                depends_build-delete \
+                                    port:ninja
+                default build.cmd   make
+                default build.post_args \
+                                    {VERBOSE=ON}
+                default destroot.target \
+                                    install/fast
+                default destroot.post_args \
+                                    "DESTDIR=${destroot}"
+                # unset the DESTDIR env. variable if it has been set before
+                if {[info exists ::env(DESTDIR)]} {
+                    unset ::env(DESTDIR)
+                }
+            }
+            Ninja {
+                ui_debug "Selecting the Ninja generator"
+                cmake.generator     Ninja
+                depends_build-append \
+                                    port:ninja
+                default build.cmd  ninja
+                default build.post_args \
+                                    ""
+                default destroot.target \
+                                    install
+                # ninja needs the DESTDIR argument in the environment
+                default destroot.post_args \
+                                    ""
+                set ::env(DESTDIR)  ${destroot}
```

Comment:
destroot.env-append I presume. I see there's a default that looks like it shouldn't be messed with.